pkgs/build-support/php/build-composer-deps.nix

```
@@ -0,0 +1,53 @@
+{ stdenvNoCC, lib, writeTextDir, php }:
+
+let
+  buildComposerDepsOverride = finalAttrs: previousAttrs:
+
+  let
+    phpDrv = finalAttrs.php or php;
+    composer = finalAttrs.composer or phpDrv.packages.composer;
+  in {
```

Will also want to ensure the mandatory attributes are specified, maybe something like

```suggestion
  in
  assert (lib.assertMsg (previousAttrs ? src) "buildComposerDeps expects src argument.");
  assert (lib.assertMsg (previousAttrs ? vendorHash) "buildComposerDeps expects vendorHash argument.");
  {
```

---

Actually, we can probably guarantee their existence by destructing the `previousAttrs`. That one should probably work without mutual recursion.

---

fixed